backend/backends/datadog/datadog.go

```diff
@@ -257,10 +261,13 @@ func (d *client) doPost(path string, body []byte) backoff.Operation {
 			return fmt.Errorf("error POSTing: %s", strings.Replace(err.Error(), d.apiKey, "*****", -1))
 		}
 		defer resp.Body.Close()
-
+		body := io.LimitReader(resp.Body, maxResponseSize)
```

Comment:
From https://golang.org/src/net/http/response.go?s=512:3205#L19 :
```
    46		// Body represents the response body.
    47		//
    48		// The http Client and Transport guarantee that Body is always
    49		// non-nil, even on responses without a body or responses with
    50		// a zero-length body. It is the caller's responsibility to
    51		// close Body. The default HTTP client's Transport does not
    52		// attempt to reuse HTTP/1.0 or HTTP/1.1 TCP connections
    53		// ("keep-alive") unless the Body is read to completion and is
    54		// closed.
    55		//
    56		// The Body is automatically dechunked if the server replied
    57		// with a "chunked" Transfer-Encoding.
    58		Body io.ReadCloser
```
We were closing the Body but Datadog sends some response (something like `{"status": "ok"}`) and we were not reading it. So the transport was not able to reuse the connection. It is very similar to how Java's URLConnection works - you have to consume the response, not just close it, otherwise it will not reuse the connection and just close it instead.

http://www.ceus-now.com/reusing-http-connections-in-golang/
https://docs.oracle.com/javase/8/docs/technotes/guides/net/http-keepalive.html